pkg/reconciler/knativeserving/knativeserving_controller.go

```diff
@@ -233,6 +225,63 @@ func (r *ReconcileKnativeServing) install(instance *servingv1alpha1.KnativeServi
 	return nil
 }
 
+func (r *ReconcileKnativeServing) executeInstall(extensions common.Extensions, instance *servingv1alpha1.KnativeServing) error {
+	err := r.config.Transform(extensions.Transform(r.scheme, instance)...)
+	if err != nil {
+		return err
+	}
+	err = extensions.PreInstall(instance)
+	if err != nil {
+		return err
+	}
+	err = r.applyAllResources(instance)
+	if err != nil {
+		return err
+	}
+	return extensions.PostInstall(instance)
+}
+
+func (r *ReconcileKnativeServing) applyAllResources(instance *servingv1alpha1.KnativeServing) error {
+	for _, spec := range r.config.Resources {
+		if !shouldApplySpec(spec, instance) {
+			continue
+		}
+
+		if err := r.config.Apply(&spec); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func shouldApplySpec(spec unstructured.Unstructured, instance *servingv1alpha1.KnativeServing) bool {
+	if instance.Spec.DisabledComponents.NetworkingIstio && isNetworkingIstioComponent(spec) {
+		return false
+	}
+	if instance.Spec.DisabledComponents.Certmanager && isCertmanagerComponent(spec) {
+		return false
+	}
+	return true
+}
+
+func isCertmanagerComponent(spec unstructured.Unstructured) bool {
+	for labelName, labelValue := range spec.GetLabels() {
+		if labelName == "networking.knative.dev/certificate-provider" && labelValue == "cert-manager" {
+			return true
+		}
+	}
+	return false
+}
+
+func isNetworkingIstioComponent(spec unstructured.Unstructured) bool {
```

Comment:
What's the reason for the different detection pattern between this and isCertmanagerComponent?

Is skiping the Deployment named "networking-certmanager" not equivalent to disabling certmanager?

Reply:
I'm curious about this, too. It'd be nice if we could omit all the irrelevant resources by label, if possible, e.g. `ClusterRole` and `ConfigMap`, but I'm not sure about the `Gateway` resources that also have this label.